Sources/AWSSDKSwiftCore/HTTP/AsyncHTTPClient.swift

```diff
@@ -18,19 +18,71 @@ import NIO
 
 /// comply with AWSHTTPClient protocol
 extension AsyncHTTPClient.HTTPClient: AWSHTTPClient {
+
+    /// write stream to StreamWriter
+    private func writeToStreamWriter(
+        writer: HTTPClient.Body.StreamWriter,
+        size: Int?,
+        on eventLoop: EventLoop,
+        stream: @escaping (EventLoop)->EventLoopFuture<ByteBuffer>) -> EventLoopFuture<Void> {
+        let promise = eventLoop.makePromise(of: Void.self)
+
+        func _writeToStreamWriter(_ amountLeft: Int?) {
+            // get byte buffer from closure, write to StreamWriter, if there are still bytes to write then call
+            // _writeToStreamWriter again.
+            _ = stream(eventLoop)
+                .map { (byteBuffer)->() in
+                    // if no amount was set and the byte buffer has no readable bytes then this is assumed to mean
+                    // there will be no more data
+                    if amountLeft == nil && byteBuffer.readableBytes == 0 {
+                        promise.succeed(())
+                        return
+                    }
+                    let newAmountLeft = amountLeft.map { $0 - byteBuffer.readableBytes }
+                    _ = writer.write(.byteBuffer(byteBuffer)).flatMap { ()->EventLoopFuture<Void> in
```

Comment:
we will need to talk to Cory here, if it reasonable to wait after every write. Or if we should write a number of bytes as fast as possible and then wait till those are written to disk. Maybe we write 16kb as fast as we can and then we wait for the buffer to be empty. otherwise with small byte sizes this might get very slow.

Reply:
You need to wait, to avoid back pressure. There is related conversation here https://github.com/swift-server/async-http-client/issues/194#issuecomment-629249604. We could collate bytebuffers until we reach a certain size, but that does complicate things considerably. Maybe for another PR?

Reply:
Okay. Later pr sounds good to me.